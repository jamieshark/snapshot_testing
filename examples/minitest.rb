require "bundler/setup"
require "minitest/autorun"
require "snapshot_testing/minitest"

class ExampleTest < Minitest::Test
  include SnapshotTesting::Minitest

  def test_snapshot
    assert_snapshot "hello"
    assert_snapshot "goodbye"
  end

  def test_named_snapshot
    assert_snapshot "named.minitest.test.txt", "named"
  end

  define_method "test_escapes_regex_[]" do
    assert_snapshot "hello"
  end
end

class ExampleSpec < Minitest::Spec
  include SnapshotTesting::Minitest

  it "takes a snapshot" do
    _("hello").must_match_snapshot
    _("goodbye").must_match_snapshot
  end

  it "takes a named snapshot" do
    _("named").must_match_snapshot "named.minitest.spec.txt"
  end

  it "escapes regex []" do
    _("hello").must_match_snapshot
  end
end
