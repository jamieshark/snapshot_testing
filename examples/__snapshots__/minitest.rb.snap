snapshots["test_0001_takes a snapshot 1"] = <<-SNAP
hello
SNAP

snapshots["test_0001_takes a snapshot 2"] = <<-SNAP
goodbye
SNAP

snapshots["test_snapshot 1"] = <<-SNAP
hello
SNAP

snapshots["test_snapshot 2"] = <<-SNAP
goodbye
SNAP

snapshots["test_0003_escapes regex [] 1"] = <<-SNAP
hello
SNAP

snapshots["test_escapes_regex_[] 1"] = <<-SNAP
hello
SNAP
